rust/src/transformations/make_stable_expr/expr_binary/mod.rs

```diff
@@ -0,0 +1,78 @@
+use polars::prelude::*;
+use polars_plan::dsl::Expr;
+use polars_plan::utils::expr_output_name;
+
+use crate::core::{Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{
+    AtomDomain, ExprContext, ExprDomain, LogicalPlanDomain, OuterMetric, SeriesDomain,
+};
+use crate::error::*;
+use crate::transformations::DatasetMetric;
+
+use super::StableExpr;
+
+#[cfg(test)]
+mod test;
+
+/// Make a Transformation that returns a binary expression
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The clipping expression
+pub fn make_expr_binary<M: OuterMetric>(
+    input_domain: ExprDomain,
+    input_metric: M,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, M, M>>
+where
+    M::InnerMetric: DatasetMetric,
+    M::Distance: Clone,
+    (ExprDomain, M): MetricSpace,
+    Expr: StableExpr<M, M>,
+{
+    let Expr::BinaryExpr { left, op, right } = expr.clone() else {
+        return fallible!(MakeTransformation, "expected binary expression");
+    };
+
+    let ExprDomain {
+        frame_domain,
+        context,
+    } = input_domain.clone();
+
+    let expr_domain = ExprDomain::new(frame_domain, ExprContext::RowByRow);
+    let t_left = left.make_stable(expr_domain.clone(), input_metric.clone())?;
+    let t_right = right.make_stable(expr_domain.clone(), input_metric.clone())?;
+
+    use polars_plan::dsl::Operator::*;
+    if !matches!(op, Eq | NotEq | Lt | LtEq | Gt | GtEq | And | Or | Xor) {
```

Comment:
I've found https://docs.rs/polars-lazy/latest/polars_lazy/dsl/enum.Operator.html, but I haven't found definitions for these. In particular, what about `LogicalAnd` and `LogicalOr`? Are they just aliases? Or are plain `And` and `Or` bitwise?